### PR TITLE
Bump dependencies. Changes to get latest c/r library working.

### PR DESCRIPTION
**Pull request by irbekrm**
This PR:
- updates go module dependencies
- bumps Go 1.19 -> 1.20
- makes changes required to use latest c/r library:
  - updates controller watch setup
  - changes to use the new webhook setup mechanism as the one we currently use has been removed
  - changes how the fake cache gets loaded with objects
  - ⚠️ BREAKING ⚠️ changes validating webhook path `/validate` -> `/validate-trust-cert-manager-io-v1alpha1-bundle` as this is [hardcoded in the new c/r webhook setup](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.15.0-alpha.0/pkg/builder/webhook.go#L247-L250). This has a potential of causing webhook downtime during upgrade to the version of trust-manager that adds this change (if the `ValidatingWebhookConfiguration` has been updated, but the webhook is still running the old version that does not yet serve to the new path)

Look at https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0 to understand the changes

This PR is also very similar to https://github.com/cert-manager/approver-policy/pull/233

### CODE REVIEW

File: pkg/bundle/controller.go [hunk 1]
```diff
@@ -95,28 +93,52 @@ func AddBundleController(ctx context.Context, mgr manager.Manager, opts Options)
...
-		}, builder.OnlyMetadata).
-
-		////// Sources //////
+		// Reconcile a Bundle on events against a ConfigMap that it
+		// owns. Only cache ConfigMap metadata.
+		WatchesMetadata(&corev1.ConfigMap{}, handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &trustapi.Bundle{}, handler.OnlyControllerOwner())).
 
 		// Reconcile trust.cert-manager.io Bundles
-		Watches(source.NewKindWithCache(new(trustapi.Bundle), sourceCache), &handler.EnqueueRequestForObject{}).
+		WatchesRawSource(&source.Informer{Informer: bundleInformer}, &handler.EnqueueRequestForObject{}).
```
Comment (irbekrm) [inline pkg/bundle/controller.go:136]: This is the new set up of how to use external cache

File: pkg/webhook/validation_test.go [hunk 2]
```diff
@@ -20,11 +20,10 @@ import (
 	"context"
 	"testing"
 
-	admissionv1 "k8s.io/api/admission/v1"
```
Comment (irbekrm) [inline pkg/webhook/validation_test.go:23]: The test change is purely in response to the new webhook setup mechanism. A couple test cases that were dealing with populating admission response/retrieving object from admission request have been removed as we no longer need to do that- c/r gives our webhook validator the decoded object from the request and only expects back an error and warnings.

File: deploy/charts/trust-manager/templates/webhook.yaml [hunk 3]
```diff
@@ -49,4 +49,4 @@ webhooks:
       service:
         name: {{ include "trust-manager.name" . }}
         namespace: {{ .Release.Namespace | quote }}
-        path: /validate
+        path: /validate-trust-cert-manager-io-v1alpha1-bundle
```
Comment (irbekrm) [inline deploy/charts/trust-manager/templates/webhook.yaml:52]: See https://github.com/cert-manager/approver-policy/pull/233#discussion_r1194138573

File: pkg/bundle/controller.go [hunk 4]
```diff
@@ -172,8 +186,8 @@ func AddBundleController(ctx context.Context, mgr manager.Manager, opts Options)
 
 		// Watch Secrets in trust Namespace. Only cache metadata.
 		// Reconcile Bundles who reference a modified source Secret.
-		Watches(source.NewKindWithCache(new(corev1.Secret), sourceCache), handler.EnqueueRequestsFromMapFunc(
-			func(obj client.Object) []reconcile.Request {
+		WatchesRawSource(&source.Informer{Informer: secretInformer}, handler.EnqueueRequestsFromMapFunc(
+			func(_ context.Context, obj client.Object) []reconcile.Request {
```
Comment (SgtCoDFish) [inline pkg/bundle/controller.go:190]: question: here we ignore the context that's passed to this function in favour of the one that's in scope (see e.g. `bundleList := b.mustBundleList(ctx)`).

Why not use this context instead? It seems possible that the controller-runtime logic which calls this function could modify the context in some way.

```suggestion
			func(ctx context.Context, obj client.Object) []reconcile.Request {
```

**Review by SgtCoDFish — COMMENTED** [review]
Couple of questions / suggestions on this - thanks for doing it!

File: pkg/bundle/controller.go (hunk 4 above)
Comment (irbekrm) [reply pkg/bundle/controller.go]: Thanks, didn't notice this one was actually using context!

**Comment by irbekrm** [comment]
Thanks for taking a look and spotting the issues @SgtCoDFish ! Think I've now fixed those

File: pkg/bundle/controller.go [hunk 5]
```diff
@@ -103,28 +93,52 @@ func AddBundleController(ctx context.Context, mgr manager.Manager, opts Options)
...
 		// Reconcile trust.cert-manager.io Bundles
-		Watches(source.NewKindWithCache(new(trustapi.Bundle), sourceCache), &handler.EnqueueRequestForObject{}).
+		WatchesRawSource(&source.Informer{Informer: bundleInformer}, &handler.EnqueueRequestForObject{}).
 
 		// Watch all Namespaces. Cache whole Namespaces to include Phase Status.
 		// Reconcile all Bundles on a Namespace change.
-		Watches(source.NewKindWithCache(new(corev1.Namespace), sourceCache), handler.EnqueueRequestsFromMapFunc(
-			func(obj client.Object) []reconcile.Request {
+		WatchesRawSource(&source.Informer{Informer: namespaceInformer}, handler.EnqueueRequestsFromMapFunc(
+			func(_ context.Context, obj client.Object) []reconcile.Request {
```
Comment (SgtCoDFish) [inline pkg/bundle/controller.go:141]: suggestion: This function also uses context by the looks of it!

```suggestion
			func(ctx context.Context, obj client.Object) []reconcile.Request {
```

File: pkg/bundle/controller.go [hunk 6]
```diff
@@ -142,8 +156,8 @@ func AddBundleController(ctx context.Context, mgr manager.Manager, opts Options)
 
 		// Watch ConfigMaps in trust Namespace. Only cache metadata.
 		// Reconcile Bundles who reference a modified source ConfigMap.
-		Watches(source.NewKindWithCache(new(corev1.ConfigMap), sourceCache), handler.EnqueueRequestsFromMapFunc(
-			func(obj client.Object) []reconcile.Request {
+		WatchesRawSource(&source.Informer{Informer: configMapInformer}, handler.EnqueueRequestsFromMapFunc(
+			func(_ context.Context, obj client.Object) []reconcile.Request {
```
Comment (SgtCoDFish) [inline pkg/bundle/controller.go:160]: suggestion: same here!

```suggestion
			func(ctx context.Context, obj client.Object) []reconcile.Request {
```

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]

File: pkg/bundle/controller.go (hunk 5 above)
Comment (irbekrm) [reply pkg/bundle/controller.go]: Thanks for spotting those!

**Comment by irbekrm** [comment]
weird docker flake

/retest

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

This looks great and this is a piece of work that would've sucked to do so I really appreciate you doing it! :rocket:

**Pull request merged**